packages/sdk/test/drive.js

```diff
@@ -7,6 +7,25 @@ import Hyperswarm from 'hyperswarm'
 import SDK, { Hyperdrive } from '../index.js'
 import { tmpdir } from './helpers/index.js'
 
+test('drive - deduplicate', async (t) => {
+  const testnet = await createTestnet(3, t.teardown)
+  const sdk = new SDK({ ...testnet, storage: tmpdir() })
+
+  const key = b4a.alloc(32).fill(0)
+
+  const a = sdk.drive(key)
+  const b = sdk.drive(key)
+
+  t.is(a, b)
+
+  a.core.closed = true
+
+  const c = sdk.drive(key)
+  t.not(c, a)
```

Comment:
so here `c` is the new drive which will overwrite internal state (`this._openedDrives`)? why not reopen existing closed drive instead?